app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/WifiDirectBroadcastReceiver.java

```diff
@@ -0,0 +1,81 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.NetworkInfo;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import org.kiwix.kiwixmobile.R;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+/**
+ * Helper class for the local file sharing module.
+ *
+ * Handles the broadcasts pertaining to the wifi p2p group formed in WiFi Direct. Works along with
+ * the wifi p2p manager in {@link LocalFileTransferActivity}.
+ */
+public class WifiDirectBroadcastReceiver extends BroadcastReceiver {
+
+  private WifiP2pManager manager;
+  private WifiP2pManager.Channel channel;
+  private LocalFileTransferActivity wifiActivity;
+
+  public WifiDirectBroadcastReceiver(@NonNull WifiP2pManager manager, @NonNull WifiP2pManager.Channel channel,
```

Comment:
It would be best if this also communicated via callbacks and only to the WifiDirectManager.

Reply:
Currently only the calls related to UI updates are made by the broadcast receiver to the LocalFileTransferActivity
Should I instead re-route them through the WifiDirectManager (BroadcastReceiver -> WifiDirectManager -> LocalFileTransferActivity)?

Reply:
Yes, absolutely

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/2e41f51fc84b3d1024ff400b7835d24a73a296af)

Reply:
This class should not contain a single reference to wifiDirectManager/WifiP2pManager/WifiP2pManager.Channel (aside from necessary constants)

Reply:
So in the constructor I should just pass-in the callbacks object?

Reply:
Yes or you can have an explicit `setListener` but constructor is best, that way the listener field can never be null

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/ba00ece2ca1b9d3cba9ee40eba1c0a2b2022c8fd)